### PR TITLE
Vector tiles: order hits by geometry size by default

### PR DESCRIPTION
**Pull request by iverase**
Vector tiles can only display polygons and lines that are big enough for the precision of the tile. This might create strange results where the hits layer of a vector tile contains no data even though the query have hits. In order to provide expected results we are ordering the hits result by geometry size so we always render polygon and lines which are big enough on respect to the tile precision.

The first issue is to define how to compute the geometry size. This size needs to computed in respect to the spherical mercator projection as it introduces distortion. In this approach we use the diagonal size of the geometry bounding in the spherical mercator projection as ordering value.

In order to perform this, we introduce the following changes:

* Introduce two new methods in our ScriptDocValue geometry interface:
** getMercatorWidth(): width of the geometry in the spherical mercator projection.
** getMercatorHeight(): Height of the geometry in the spherical mercator projection.

* Move `SphericalMercatorUtil` to server under common/geo.

Then if there is no ordering provided by the user when calling the `_mvt` API and the size of the request > 0, then we add a order by using a script that computes the diagonal of the geometry square. This is used to order the results.

Note: We have consider to use filtering but this would affect the results on the aggregation.

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

**Review by jdconrad — APPROVED** [review]
LGTM from the scripting side

**Review by imotov — COMMENTED** [review]
As I mentioned before I think it would have been much better if we had a dedicated sort for that. But considering that this is a somewhat exotic feature we can probably go with the script for now, but we shouldn't generate this script on the fly.

File: x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java [hunk 1]
```diff
@@ -317,7 +317,28 @@ private void setAggBuilder(AggregatorFactories.Builder aggBuilder) {
...
+        if (sortBuilders == null) {
+            if (size == 0) {
+                // no need to add sorting
+                return List.of();
+            }
+            return List.of(
+                new ScriptSortBuilder(
+                    new Script(
+                        "double w = doc[\""
+                            + getField()
```
[the remark this answers is not in the document]
Comment (iverase) [reply x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java]: done

[the remark this answers is not in the document]
Comment (jdconrad) [reply x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileRequest.java]: Yes! Good catch. I see @iverase has already incorporated this suggestion :+1:

**Review by imotov — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**